### PR TITLE
syntax: strip escaped newlines from all non-quoted words

### LINKED ISSUE #832: syntax: the printer can split up strings when joined with escaped newlines
I came across this kind of assignment in a legacy codebase:

```sh
# Has the same effect as: wrapped='a b c'
wrapped=\
"a "\
"b "\
"c"
```

which `shfmt` turns into:
```sh
wrapped="a "\
	"b "\
	"c"
```

and causes the runtime error: `line 4: b : command not found`.

Tested on `GNU bash, version 5.1.4(1)-release (x86_64-pc-linux-gnu)` and the latest `shfmt`.

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

Fixes #832.

### CODE REVIEW

**Comment by mvdan** [comment]
cc @cilki

File: syntax/printer.go [hunk 1]
```diff
@@ -583,16 +583,15 @@ func (p *Printer) wordParts(wps []WordPart, quoted bool) {
 		if i+1 < len(wps) {
 			next = wps[i+1]
 		}
+		// Keep escaped newlines separating word parts when quoted.
+		// Note that those escaped newlines don't cause indentaiton.
```
Comment (theclapp) [inline syntax/printer.go:587]: "indentaiton"

Comment (mvdan) [reply syntax/printer.go:587]: thainks :)

**Comment by mvdan** [comment]
Thanks both for the reviews! Merging.

**Pull request merged**